src/Artsy/Router/buildServerApp.tsx

```diff
@@ -81,23 +83,34 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
           )
         }
 
-        const { loadableState, relayData: _relayData, styleTags } = await trace(
+        const {
+          loadableState,
+          relayData: _relayData,
+          styleTags,
+          bodyHTML,
+        } = await trace(
           "buildServerApp.fetch",
           (async () => {
             // Kick off relay requests to prime cache. TODO: Remove the need to
             // do this by using persisted queries.
-            ReactDOMServer.renderToString(<ServerApp />)
```

Comment:
Each time we render our tree _without_ wrapping it in the SC context, we’re rendering using the StyleSheet master singleton, which _injects_ (aka caches) the rules [in its state](https://github.com/styled-components/styled-components/blob/v4.0.3/src/models/StyleSheet.js#L290).

Reply:
This is pretty non-obvious... I wonder if we should open a PR back to styled-components that updates the **NOTE:** section? https://www.styled-components.com/docs/advanced#server-side-rendering

Reply:
That being said, us having to `renderToString` twice in order to prime our relay cache feels like a somewhat unique circumstance.

Reply:
I do feel like there’s probably the opportunity to improve the docs to make it very clear that you always want to render in a separate stylesheet during SSR.